lib/elixir/lib/module/types/unify.ex

```diff
@@ -834,6 +922,18 @@ defmodule Module.Types.Unify do
     "var#{index + 1}"
   end
 
+  def format_type({:fun, clauses}, simplify?) do
+    format =
+      Enum.map_join(clauses, "; ", fn {params, return} ->
+        params = Enum.map_join(params, ", ", &format_type(&1, simplify?))
+        params = if params == "", do: params, else: "#{params} "
+        return = format_type(return, simplify?)
+        "#{params}-> #{return}"
+      end)
+
+    "(#{format})"
+  end
```

Comment:
For this function, perhaps build iodata and call iodata_to_binary at the end - or let's even change the definition and the callers of this function to allow iodata as return.